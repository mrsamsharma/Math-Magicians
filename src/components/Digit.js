import React from 'react';
import PropTypes from 'prop-types';

export default function Digit({ digit, eventHandle }) {
  return (
    <button type="button" onClick={eventHandle}>{digit}</button>
  );
}

Digit.propTypes = {
  digit: PropTypes.string,
  eventHandle: PropTypes.func,
};

Digit.defaultProps = {
  digit: '1',
  eventHandle: { },
};
