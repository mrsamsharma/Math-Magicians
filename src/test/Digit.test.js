import React from 'react';
import Digit from '../components/Digit';
import renderer from 'react-test-renderer';

it('Digit Component renders correctly', () => {
    const tree = renderer
      .create(<Digit />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });