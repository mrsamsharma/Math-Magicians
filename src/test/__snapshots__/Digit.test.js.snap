// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Digit Component renders correctly 1`] = `
<button
  onClick={Object {}}
  type="button"
>
  1
</button>
`;
